modules/siddhi-core/src/main/java/io/siddhi/core/table/CacheTable.java

```diff
@@ -0,0 +1,310 @@
+                                                                    storeVariableExpressionExecutors) {
+        boolean routeToCache = checkConditionToRouteToCache(condition);
+        MetaStateEvent metaStateEvent = new MetaStateEvent(storeMatchingMetaInfoHolder.getMetaStateEvent().
+                getMetaStreamEvents().length);
+        for (MetaStreamEvent referenceMetaStreamEvent: storeMatchingMetaInfoHolder.getMetaStateEvent().
+                getMetaStreamEvents()) {
+            metaStateEvent.addEvent(referenceMetaStreamEvent);
+        }
+        MatchingMetaInfoHolder matchingMetaInfoHolder = new MatchingMetaInfoHolder(
+                metaStateEvent,
+                storeMatchingMetaInfoHolder.getMatchingStreamEventIndex(),
+                storeMatchingMetaInfoHolder.getStoreEventIndex(),
+                storeMatchingMetaInfoHolder.getMatchingStreamDefinition(),
+                this.tableDefinition,
+                storeMatchingMetaInfoHolder.getCurrentState());
+
+        Map<String, Table> tableMap = new ConcurrentHashMap<>();
+        tableMap.put(this.tableDefinition.getId(), this);
+
+        return new CacheCompiledConditionWithRouteToCache(compileCondition(condition, matchingMetaInfoHolder,
+                storeVariableExpressionExecutors, tableMap, siddhiQueryContext, routeToCache), routeToCache);
+    }
+
+    /**
+     * wrapper to send routeToCache bool with cache compiled condition
+     */
+    public class CacheCompiledConditionWithRouteToCache {
+        private CompiledCondition cacheCompiledCondition;
+        private boolean routeToCache;
+
+        CacheCompiledConditionWithRouteToCache(CompiledCondition cacheCompiledCondition, boolean routeToCache) {
+            this.cacheCompiledCondition = cacheCompiledCondition;
+            this.routeToCache = routeToCache;
+        }
+
+        public CompiledCondition getCacheCompiledCondition() {
+            return cacheCompiledCondition;
+        }
+
+        public boolean isRouteToCache() {
+            return routeToCache;
+        }
+    }
+
+    private boolean checkConditionToRouteToCache(Expression condition) {
+        List<String> primaryKeysArray = new ArrayList<>();
+        Annotation primaryKeys = getAnnotation("PrimaryKey", tableDefinition.getAnnotations());
+        if (primaryKeys == null) {
+            return false;
+        }
+        List<Element> keys = primaryKeys.getElements();
+        for (Element element: keys) {
+            primaryKeysArray.add(element.getValue());
+        }
+        recursivelyCheckConditionToRouteToCache(condition, primaryKeysArray);
+        return primaryKeysArray.size() == 0;
+    }
+
+    private void recursivelyCheckConditionToRouteToCache(Expression condition, List<String> primaryKeysArray) {
+        if (condition instanceof Compare) {
+            Compare compareCondition = (Compare) condition;
+            if (compareCondition.getOperator().name().equalsIgnoreCase("EQUAL")) {
```

Comment:
This is an enum so just use == check for the enum and do not use `.name().equalsIgnoreCase("EQUAL")`